Code/Engine/GameEngine/Messages/DamageMessage.h

```diff
@@ -1,19 +1,22 @@
 #pragma once
 
+#include <Core/Messages/EventMessage.h>
+#include <Foundation/Communication/Message.h>
 #include <GameEngine/GameEngineDLL.h>
 #include <GameEngine/VisualScript/VisualScriptNode.h>
-#include <Foundation/Communication/Message.h>
 
-struct EZ_GAMEENGINE_DLL ezMsgDamage : public ezMessage
+struct EZ_GAMEENGINE_DLL ezMsgDamage : public ezEventMessage
 {
-  EZ_DECLARE_MESSAGE_TYPE(ezMsgDamage, ezMessage);
+  EZ_DECLARE_MESSAGE_TYPE(ezMsgDamage, ezEventMessage);
 
   double m_fDamage = 0;
+  ezString m_sHitObjectName; ///< The actual game object that was hit (may be a child of the object to which the message is sent)
```

Comment:
Why the name and not a gameobject handle?

Reply:
So right now the script integration is not able to handle GameObject and Component handles, because for that the ezVariant requires an upgrade, which Christopher wanted to do at some point. Therefore instead of hacking anything in there to make handles work, I am just waiting for the variant change and may change this message then as well.

On the other hand, it also seems to be convenient to use a string here, because this way you can reuse the same name for objects that should have the same effect (multiple hit boxes for a complex object). However, that is probably not such a common use case, so wouldn't mind to change it again.